### DISCUSSION TITLE
In-line meter connected via Home Network/Energynet

### DISCUSSION THREAD

**Discussion by d0minander** [WillCodeForCats/solaredge-modbus-multi#686] [General]
Hello,

I have an inverter with the latest firmware and moved my Meter from RS485 to Home Network, now i can no longer see meter data, i can of course move it back to RS485. but i think it would be nice to be able to use a wireless meter in the future.

I have seen info about this issue in the Wiki but also a statement it should be fixed in a software version that has now been released.
Is the problem still that the Inverter does not supply the info about a meter connected via Home Network via modbus?
Is there anything we can to do help in terms of supplying you data, or should we contact SolarEdge about it?

Thanks for a great tool!
//Niklas

**Comment by WillCodeForCats** [comment]
Support has to come from the SolarEdge side.

**Comment by d0minander** [comment]
Understood, so not much i can do to help other then file a support ticket to SolarEdge and ask for them to add modbus capabilities for meters connected via Home Network?

**Comment by Zandor300** [comment]
@d0minander
SolarEdge Inline Meters should be exposed through SunSpec when running firmware version 4.18.32 or newer, though I've seen that it can be slightly flaky.

The latest firmware version for residential inverters is 4.21.23. Your installer can update the firmware remotely if you ask them. Otherwise, create a ticket with SolarEdge asking them to update your inverter firmware.

**Before you ask to update, please check which version you are running and report back so we can see if it's an issue with the firmware!**

How to check:  
Open monitoring.solaredge.com on desktop > Go to your installation > Layout > Rightclick your inverter > Info.  
In the top section of that dialog, it should mention something like CPU. Please report that back here so we know.

*SolarEdge firmware release notes can be found [here](https://www.solaredge.com/en/support/setapp-inverters-firmware) fyi.*

**Comment by d0minander** [comment]
Hello @Zandor300 
I have already updated the inverter to 4.21.23
They also updated the Inline meter to 1.081 and the Home Network module to 1.1.3
So all firmwares are the latest, but i still cant see any meters, i tried to remove and re-add the inverter in Home Assistant but still no meter.
So i will move back to RS485 when i have time.
Also in a few weeks im getting the SolaEdge EV charger (three phase), so hopefully that is soon supported, otherwise it seems like it can be added via the Keba integration since its a rebranded Keba P30 that has modbus capability.

**Comment by Zandor300** [comment]
@d0minander Have you tried restarting the inverter? Sometimes that will help I've noticed.

_Don't know how experienced you are with SolarEdge but:_  

- Turn P/1/0 switch to 0 and wait until green LED starts blinking
- AC off
- Check that LEDs are off
- AC on
- P/1/0 switch back to 1

Turning of PV production before cutting AC connection is actually very important since I've heard some horror stories of the inverter blowing because the customer shut down the inverter too many times by cutting AC without first stopping PV production.

**Comment by d0minander** [comment]
I do belive i have tried restarting it, but i will give it another try.
I am fairly experienced wirth Solaredge, im used to using SetApp to upgrade firmware and configure things, like meters.
One thing to note is that i have an SE17K that initially only supported commercial firmware, SolarEdge now officially support residential firmware on SE16K and SE17K also, but then support needs to do the upgrade.

**Comment by d0minander** [comment]
@Zandor300 Restart did not help, the meter is not getting found with all the latest firmwares.
I tried to remove the device and re-add it also, still no meter discovered.

Will change back to RS485 when i have some time.

**Comment by d0minander** [comment]
Reconfigured the meter to RS485 and now i can see it again in home assistant.
So i would say its confirmed that Home Network does still not work.

**Comment by d0minander** [comment]
Hello, another update on this topic that might be interesting.
I did another test moving the meter to Home Network connection instead of RS485, this time i did not remove the inverter and meter from the integration before doing the move, and to my surprise the meter continued to work in home assistant after the change.

Last time i removed the inverter and meter first and then re-added them when the meter was connected via Home Network, and then it did not find the meter.

**Reply by manuschillerdev** [reply]
So my best bet if I want to use the inline meter via my home network right is to install it via RS485, let the integration detect the meter, and then switch it to use the home network?

**Reply by d0minander** [reply]
Thats what worked for me, have you tried already with the home network connection?

**Reply by manuschillerdev** [reply]
No, I am trying to decide which meter to get and therefore checking compatibility with this extension.

A wireless connection would be nice in terms of warranty, because I wouldn't have to open the Inverter
